File: php/1920_Build_Array_from_Permutation.php
<?php

class Solution
{

    /**
     * @param Integer[] $nums
     * @return Integer[]
     */
    function buildArray($nums)
    {
        $result = [];

        foreach ($nums as $key => $num) {
            $result[] = $nums[$nums[$key]];
        }

        return $result;
    }

}

File: scala/1920_Build_Array_from_Permutation.scala
object Solution {
    def buildArray(nums: Array[Int]): Array[Int] = {
        var result = new Array[Int](nums.length)
        for (i <- 0 to (nums.length - 1)) {
            result(i) = nums(nums(i))
        }
        return result
    }
}
